numpy/core/_add_newdocs.py

```diff
@@ -5491,6 +5491,39 @@
 
     """))
 
+add_newdoc('numpy.core.multiarray', 'dtype', ('metadata',
+    """
+    Either ``None`` or a readonly dictionary of metadata (mappingproxy).
+
+    The metadata field can be set using any dictionary at data-type
+    creation. Note that whether or not operations on arrays with metadata
+    attached to their datatypes is currently not well defined and should
+    not be relied on.
+
+    Examples
+    --------
+
+    >>> dt = np.dtype(float, metadata={"key": "value"})
+    >>> dt.metadata["key"]
+    'value'
+    >>> arr = np.array([1, 2, 3], dtype=dt)
+    >>> arr.dtype.metadata
+    mappingproxy({'key': 'value'})
+
+    Some operations may preserve metadata (identical data types):
```

Comment:
```suggestion
    Adding arrays with identical datatypes preserves the metadata:
```

Reply:
I think the point is that we are not trying to specify exactly which operations preserve metadata

Reply:
> I think the point

Again, it might save time if we allow the OP to weigh in first.

Reply:
Yeah, I was trying to be fuzzy, although, in this case I think we might as well define it as the correct output (unless we decide to always drop metadata, it seems always right – no matter what the propagation rules are).

Reply:
We can't write the doc without knowing what policy we want to set. Are we writing it just to say "If you use this, bend over and kiss your ass goodbye"?  Or are there certain limited uses we plan to support (and test)?